src/workload.rs

```diff
@@ -257,8 +259,26 @@ impl WorkloadManager {
         config: config::Config,
         metrics: Arc<Metrics>,
         awaiting_ready: readiness::BlockReady,
+        cert_manager: Box<dyn CertificateProvider>,
     ) -> anyhow::Result<WorkloadManager> {
-        let workloads: Arc<Mutex<WorkloadStore>> = Arc::new(Mutex::new(WorkloadStore::default()));
+        let (tx, mut rx) = watch::channel(Identity::default());
+        tokio::spawn(async move {
+            while rx.changed().await.is_ok() {
+                let workload_identity = rx.borrow().clone();
+                match cert_manager.fetch_certificate(&workload_identity).await {
+                    Ok(_) => debug!("prefetched cert for {:?}", workload_identity.to_string()),
+                    Err(e) => error!(
+                        "unable to prefetch cert for {:?}, skipping, {:?}",
+                        workload_identity.to_string(),
+                        e
+                    ),
+                }
+            }
+        });
+        let workloads: Arc<Mutex<WorkloadStore>> = Arc::new(Mutex::new(WorkloadStore {
+            cert_tx: Some(tx),
```

Comment:
Do we need to make it an Option? Or can we just store directly?

Reply:
I made it an option so I didn't have to deal with the default macro, lmk if theres a workaround

Reply:
Alternative would be to drop Default, make a `WorkloadStore::new() -> (WorkloadStore, rx)`. In new manually set the ~5 hashmaps. But this is ok too